bundles/org.openhab.binding.roborock/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,828 @@
+		<label>History</label>
+		<channels>
+			<channel id="total-clean-area" typeId="total-clean-area"/>
+			<channel id="total-clean-time" typeId="total-clean-time"/>
+			<channel id="total-clean-count" typeId="total-clean-count"/>
+		</channels>
+	</channel-group-type>
+	<channel-group-type id="info">
+		<label>Info</label>
+		<channels>
+			<channel id="multi-maps-list" typeId="multi-maps-list"/>
+			<channel id="room-mapping" typeId="room-mapping"/>
+			<channel id="routine-mapping" typeId="routine-mapping"/>
+			<channel id="fw-features" typeId="fw-features"/>
+			<channel id="customize-clean-mode" typeId="customize-clean-mode"/>
+			<channel id="carpet-mode" typeId="carpet-mode"/>
+		</channels>
+	</channel-group-type>
+	<channel-group-type id="network">
+		<label>Network</label>
+		<channels>
+			<channel id="ssid" typeId="ssid"/>
+			<channel id="bssid" typeId="bssid"/>
+			<channel id="rssi" typeId="rssi"/>
+		</channels>
+	</channel-group-type>
+
+	<!-- Status channels -->
+	<channel-type id="clean-area">
+		<item-type>Number:Area</item-type>
+		<label>Cleaning Area</label>
+		<tags>
+			<tag>Status</tag>
+			<tag>Info</tag>
+		</tags>
+		<state pattern="%.2f %unit%" readOnly="true"/>
+	</channel-type>
+	<channel-type id="clean-time">
+		<item-type>Number:Time</item-type>
+		<label>Cleaning Time</label>
+		<tags>
+			<tag>Status</tag>
+			<tag>Duration</tag>
+		</tags>
+		<state pattern="%.0f %unit%" readOnly="true"/>
+	</channel-type>
+	<channel-type id="dnd-enabled">
+		<item-type>Switch</item-type>
+		<label>Do Not Disturb</label>
+		<tags>
+			<tag>Switch</tag>
+			<tag>Mode</tag>
+		</tags>
+	</channel-type>
+	<channel-type id="error-code">
+		<item-type>String</item-type>
+		<label>Error Code</label>
+		<tags>
+			<tag>Status</tag>
+			<tag>Info</tag>
+		</tags>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="error-id">
+		<item-type>Number</item-type>
+		<label>Error ID</label>
+		<tags>
+			<tag>Status</tag>
+			<tag>Info</tag>
+		</tags>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="fan-power">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Fan Power</label>
+		<tags>
+			<tag>Setpoint</tag>
+			<tag>Speed</tag>
+		</tags>
+		<state min="1" max="99" step="1" pattern="%.0f%%" readOnly="false"/>
+	</channel-type>
+	<channel-type id="in-cleaning" advanced="true">
+		<item-type>Number</item-type>
+		<label>In Cleaning?</label>
+		<description>1 = Cleaning, 0 = Not Cleaning</description>
+		<tags>
+			<tag>Status</tag>
+			<tag>Info</tag>
+		</tags>
+		<state pattern="%.0f" readOnly="true"/>
+	</channel-type>
+	<channel-type id="map-present" advanced="true">
+		<item-type>Number</item-type>
+		<label>Map Present</label>
+		<category>Energy</category>
+		<tags>
+			<tag>Status</tag>
+			<tag>Info</tag>
+		</tags>
+		<state pattern="%.0f" readOnly="true"/>
+	</channel-type>
+	<channel-type id="msg-seq" advanced="true">
+		<item-type>Number</item-type>
+		<label>Msg Seq</label>
```

Comment:
This is a bit cryptic. I think many channels can benefit from a description. You might use AI to save a lot of time to generate these, but you have to read them carefully if you do so.

Reply:
This one is so cryptic that even the python-roborock folks don't have a definition.... https://python-roborock.readthedocs.io/en/latest/api_commands.html

Reply:
I think it's just a message sequence number, hardly worth a channel, but miio has it, so I kept it.

Reply:
Added a description for this and some of the other more obscure channels.